eo-maven-plugin/src/main/java/org/eolang/maven/tojos/ForeignTojo.java

```diff
@@ -92,6 +104,63 @@ public String shortDescription() {
         );
     }
 
+    /**
+     * The tojo hash.
+     * @return The hash.
+     */
+    public String hash() {
+        return this.delegate.get(ForeignTojos.Attribute.HASH.key());
+    }
+
+    /**
+     * Checks if tojo was already optimized.
+     *
+     * @return True if optimization is required, false otherwise.
+     */
+    public boolean optimizationRequired() {
+        final Path src = this.xmir();
+        boolean res = true;
+        if (this.delegate.exists(ForeignTojos.Attribute.XMIR_2.key())) {
+            final Path tgt = this.xmirSecond();
+            if (tgt.toFile().lastModified() >= src.toFile().lastModified()) {
+                Logger.debug(
+                    this, "Already optimized %s to %s",
+                    new Rel(src), new Rel(tgt)
+                );
+                res = false;
+            }
+        }
+        return res;
+    }
+
+    /**
+     * Check if the given tojo has not been parsed.
+     *
+     * @return True if the tojo has not been parsed.
+     */
+    public boolean isNotParsed() {
```

Comment:
@volodya-lombrozo Can we make this name simpler?  According to: https://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html. Maybe `notParsed` or smth else

Reply:
@Graur Done, thanks.